compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -1325,7 +1325,7 @@ object Trees {
             this(this(x, arg), annot)
           case Thicket(ts) =>
             this(x, ts)
-          case _ if ctx.mode.is(Mode.Interactive) =>
+          case _ =>
```

Comment:
Proposal: replace with 

    case _ if ctx.errorsReported || ctx.mode.is(Mode.Interactive) =>

and adapt the comment to say that in interactive mode, errors might come from previous runs.